src/rules/__tests__/no-expect-to-match-object.test.ts

```diff
@@ -0,0 +1,23 @@
+import { TSESLint } from '@typescript-eslint/experimental-utils';
+import rule from '../no-expect-to-match-object';
+
+const ruleTester = new TSESLint.RuleTester();
+
+ruleTester.run('no-expect-to-match-object', rule, {
+  valid: [
+    'expect({ a: 1, b: 2 }).toStrictEqual({ a: 1, b: 2 });',
+    'expect({ a: 1, b: 2 }).objectContaining({ a: expect.any(Number), b: expect.any(Number) });',
+  ],
+  invalid: [
```

Comment:
Could you chuck in some invalid cases for when:

* `toMatchObject` is expected but not actually called
* `toMatchObject` is expected with no arguments
* `toMatchObject` is expected with multiple arguments
* expect is called without a value
* expect is called with a non-object value
* a few of these tests with modifiers

Reply:
Added tests in https://github.com/jest-community/eslint-plugin-jest/pull/545/commits/0f8ada9303b31b3e7936595529614921d90c64f4